lib/tools/adb-commands.js

```diff
@@ -770,6 +772,22 @@ methods.isAnimationOn = async function () {
                 (setting) => setting !== '0.0');
 };
 
+/**
+ * Change the locale on the device under test. Don't need to reboot the device after changing the locale.
+ * This method sets an arbitrary locale following:
+ *   https://developer.android.com/reference/java/util/Locale.html
+ *   https://developer.android.com/reference/java/util/Locale.html#Locale(java.lang.String,%20java.lang.String)
+ *
+ * @param {string} language - Language. e.g. en, ja
+ * @param {string} country - Country. e.g. US, JP
+ */
+methods.setDeviceSysLocaleViaSettingApp = async function (language, country) {
```

Comment:
I was thinking about some approach, which would allow us to isolate country/language selection logic on appium-adb module level instead of changing it on driver level. For example we call something like `adb.setDeviceLocale` and the internal logic automatically decides which method selection to apply based on API level, so we don't have to tune that for each driver.
How do you think @imurchie ?

Reply:
👍 I'm also thinking so (the driver just call `adb.setDeviceLocale` for example) when I read codebase and added this method.

Reply:
you can implement this approach, so then we could increase major module version and update client drivers when it is published